### PR TITLE
UI: Stateful buttons, aria-pressed and engageable-interface

### PR DESCRIPTION
**Pull request by nhaagen**
This implements an Engageable-interface for certain buttons; see https://github.com/ILIAS-eLearning/ILIAS/blob/trunk/src/UI/ROADMAP.md#engaged-buttons-advanced-4h

An engageable button can be configured ->withEngagedState(bool $state) and is rendered with a class "engaged" and an aria-pressed-attribute accordingly.

Also, the close-button implements Clickable (which it should, I think).

### CODE REVIEW

File: src/UI/Component/Button/Engageable.php
Comment (Amstutz) [inline src/UI/Component/Button/Engageable.php:12]: This is a rule that MUST be added to the rules of the parent button. Since engaged is not available on all buttons, you may rephrase this to something like: "Some Buttons can be stateful; when ...."

Comment (Amstutz) [reply src/UI/Component/Button/Engageable.php:12]: Also, some phrase in the effect part of buttons should hint to this as well. Lastly add an example showing the use of this engaged state on standard buttons.

**Review by Amstutz — CHANGES_REQUESTED** [review]
LGTM in General, however note that this is a change of interface, so we need JF approval for this. I recommend the JF to accept this proposal with two additional requests:
* [ ] adapt the effect description of buttons (see comment bellow).
* [ ] add a new rule for buttons (see comment bellow).

Before merging:
* [ ] add an example showing the new state.

**Comment by Amstutz** [comment]
This is related to: https://github.com/ILIAS-eLearning/ILIAS/pull/669

**Comment by matthiaskunkel** [comment]
Jour Fixe, 19 NOV 2018 : We highly appreciate this PR and accept it for trunk. Please merge it (after removing some typos).

**Comment by klees** [comment]
@Amstutz May I merge this or is there anything left to do from your perspective?

**Pull request merged**
